extractor/restriction_parser.cpp

```diff
@@ -140,11 +141,11 @@ RestrictionParser::TryParse(const osmium::Relation &relation) const
     }
 
     bool is_only_restriction = false;
+    const char RESTRICTION_PREFIX[] = {"restriction"};
 
     for (auto iter = fi_begin; iter != fi_end; ++iter)
     {
-        if (std::string("restriction") == iter->key() ||
-            std::string("restriction::hgv") == iter->key())
+        if (boost::algorithm::starts_with(iter->key(), RESTRICTION_PREFIX))

```

Comment:
I _think_ this check amd the `RESTRICTION_PREFIX` is redundant as the `KeyPrefixFilter` above already filters out everything that starts with `restriction`.